worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -47,6 +47,26 @@ function onBeforeProcessDocument(e) {
     e.cancel = ! eval(action.conditionFunction)(e.document, arguments);
 }
 
+function onProcessDocument(e) {
+    if (!e.application.getInputMessageProcessor().getProcessSubdocumentsSeparately()) {
+        traverseDocumentForSettingWorkerVersion(e.document);
+    } else {
+        setWorkerVersion(e.document);
+    }
+}
+
+function traverseDocumentForSettingWorkerVersion(document) {
+    setWorkerVersion(document);
+    for each(var subdoc in document.getSubdocuments()) {
+        setWorkerVersion(subdoc);
+    }
+}
+
+function setWorkerVersion(document) {
+    var versionFieldName = "PROCESSING_" + getCurrentWorkerName(document) + "_VERSION";
```

Comment:
We should make this all upper-case.  I.e. `PROCESSING_WORKER-EE_VERSION` rather than `PROCESSING_worker-ee_VERSION`.

Reply:
Done